rusty_pizza_server/src/order_model/meals.rs

```diff
@@ -31,11 +31,33 @@ impl Meals {
         self.meals.insert(id, meal);
         self.meals.get_mut(&id).unwrap()
     }
+
+    pub fn set_paid(&mut self, paid: Money) {
+        self.paid = paid;
+    }
+
+    pub fn set_tip(&mut self, tip: Money) {
+        self.tip = tip;
+    }
+
+    pub fn calculate_total_price(&self) -> Money {
+        let mut total_price = Money::new(0, 0);
+        for (_, meal) in self.meals.iter() {
+            total_price = total_price + meal.get_price();
+        }
+        return total_price;
+    }
+
+    pub fn calculate_change(&self) -> Money {
+        self.paid - self.calculate_total_price() - self.tip
```

Comment:
The `Money` type is designed to represent a positive amount of money. This calculation can easily underflow if the user did not pay enough. We need to handle this case and should thus introduce an error for this that carries the amount of money the user still has to pay.

Reply:
I have added a check to return an Error, if the `paid` Money is less than the `total_price`+`tip`.
I created #18 to extend the Money type with Comparison Operators

Reply:
I would prefer to further enrich this by returning an instance of an error enum carrying the amount of money that is missing.